### PR TITLE
[FLOC-1272] Pin an (as yet) unreleased Docker version dependency for Fedora 20

### PR DESCRIPTION
**Pull request by wallrj**
Fixes https://clusterhq.atlassian.net/browse/FLOC-1272

### CODE REVIEW

**Comment by wallrj** [comment]
The acceptance tests have now passed, despite what the build report says above.

And the Centos and Ubuntu packages have both built successfully.
 
* http://build.clusterhq.com/boxes-flocker?branch=scratch-FLOC-1272&num_builds=10

**Comment by wallrj** [comment]
Hmm. I guess this is going to prevent installation of `clusterhq-flocker-node` on Fedora20....unless we explicitly install from koji in the cloud installation instructions too :cry:

File: flocker/acceptance/test_environment_variables.py [hunk 1]
```diff
@@ -254,3 +254,5 @@ def verify_data_moves(connection_2):
 
         verifying_data_moves = getting_mysql_2.addCallback(verify_data_moves)
         return verifying_data_moves
+
+    test_moving_data.timeout = 10*60
```
Comment (adamtheturtle) [inline flocker/acceptance/test_environment_variables.py]: Seems unrelated.

**Comment by wallrj** [comment]
Yeah, looks like we'd need to update all our cloud install instructions too

```
[root@54.149.205.75] out: Error: Package: clusterhq-flocker-node-0.3.2-1.1323.g8663686.noarch (clusterhq-build)
[root@54.149.205.75] out:            Requires: docker-io = 1.4.1-8.fc20
[root@54.149.205.75] out:            Available: docker-io-1.4.1-6.fc20.x86_64 (updates)
[root@54.149.205.75] out:                docker-io = 1.4.1-6.fc20
[root@54.149.205.75] out:  You could try using --skip-broken to work around the problem
[root@54.149.205.75] out:  You could try running: rpm -Va --nofiles --nodigest
[root@54.149.205.75] out: 
```

...or @tomprince adds the required docker package to our clusterhq repo....I guess that's the plan.

File: flocker/acceptance/test_environment_variables.py (hunk 1 above)
Comment (wallrj) [inline flocker/acceptance/test_environment_variables.py]: @tomprince You were getting test timeouts, right?

**Comment by tomprince** [comment]
The branch wasn't really intended to be merged. It was just here to get buildbot to run.

File: flocker/acceptance/test_environment_variables.py (hunk 1 above)
Comment (tomprince) [inline flocker/acceptance/test_environment_variables.py]: http://build.clusterhq.com/builders/flocker%2Facceptance%2Fvagrant%2Ffedora-20/builds/384

Comment (wallrj) [inline flocker/acceptance/test_environment_variables.py]: The tests all passed with your new docker package. http://build.clusterhq.com/builders/flocker%2Facceptance%2Fvagrant%2Ffedora-20/builds/396 so perhaps this isn't strictly necessary here.

**Comment by wallrj** [comment]
We discussed in zulip and decided that this branch *should* be merged, but in such a way as to force a docker upgrade for those users who already have the broken version installed.

File: admin/packaging.py [hunk 2]
```diff
@@ -190,6 +190,17 @@ def format(self, package_type):
...
+# This ensures that servers with the broken docker-io-1.4.1 package get
+# upgraded when Flocker is installed.
+# See https://bugzilla.redhat.com/show_bug.cgi?id=1185423
+# The working 1.4.1-8 package is temporarily being hosted in the ClusterHQ
+# repo, but will soon be backported to Fedora20.
+# See https://admin.fedoraproject.org/updates/docker-io-1.4.1-8.fc20
+# In future this specific minimum version dependency can be removed.
+# See https://clusterhq.atlassian.net/browse/FLOC-1293
+FedoraDockerDependency = partial(
+    Dependency, package='docker-io', compare='>=', version='1.4.1-8.fc20')
```
Comment (adamtheturtle) [inline admin/packaging.py]: It would be nice to have a test to confirm that https://bugzilla.redhat.com/show_bug.cgi?id=1185423 is not done so we could update FLOC-1293 as soon as possible :laughing:

**Comment by adamtheturtle** [comment]
From the Flocker dev environment, installing the packages Buildbot has built from this branch updates the docker-io package (as seen from the Release output from `yum info docker-io`).

**Comment by adamtheturtle** [comment]
Requiring a newer version of Docker seems like it should go in "What's New" as well as NEWS. I'm not sure that we're meant to be updating NEWS in branches, but I think we should be updating a "Next Release" section of What's New.

**Comment by adamtheturtle** [comment]
This looks good, please update the files as suggested above and merge when Buildbot passes.

**Comment by wallrj** [comment]
Seems like too much of an implementation detail to be included on the "What's New" page. We certainly didn't update the page last time we had to bump our Docker version requirement.  It can be added to NEWS by whoever manages the next release.

I closed https://clusterhq.atlassian.net/browse/FLOC-1293 because I'm not sure we will be able to remove this Fedora20 special case. 
The special case can be removed when we next need to bump the Docker version requirement (as opposed to the Docker package release version requirement) at which point we will only need to specify the major.minor.patch version (not the specific package release version).
Hope that makes sense.

I'll merge this.
